Basic/APIDiorama/Assets/NetvarVsRpc/Scripts/ColorManager.cs

```diff
@@ -0,0 +1,93 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the color of a Networked object
+    /// </summary>
+    public class ColorManager : NetworkBehaviour
+    {
+        [SerializeField]
+        bool m_UseNetworkVariableForColor;
+
+        NetworkVariable<Color32> m_NetworkedColor = new NetworkVariable<Color32>();
+        Material m_Material;
+
+        void Awake()
+        {
+            m_Material = GetComponent<Renderer>().material;
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    OnClientColorChanged(m_Material.color, m_NetworkedColor.Value);
+                    m_NetworkedColor.OnValueChanged += OnClientColorChanged;
+                }
+            }
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    m_NetworkedColor.OnValueChanged -= OnClientColorChanged;
+                }
+            }
+        }
+
+        void Update()
+        {
+            if (!IsClient)
+            {
+                return;
+            }
+
+            bool playerIsCloseEnough = true; //todo: implement behaviour
+            if (playerIsCloseEnough)
+            {
+                if (Input.GetKeyUp(KeyCode.E))
+                {
+                    OnClientRequestColorChange();
+                }
+            }
+        }
+
+        void OnClientRequestColorChange()
+        {
+            OnServerChangeColorServerRpc();
+        }
+
+        [ServerRpc]
```

Comment:
Alternatively, this Rpc could also be done on the PlayerManager instead

Reply:
> By default, ServerRpcs require to be called from a NetworkObject that is owned by the client, so here since the colored cubes are owned by the server, we get an error when trying to call this rpc as a client that is not the host.

Quick clarification, that is owned by the client, not the NetworkObject. If Client 1 owns NetworkObject_A, NetworkObject_B on that same client can call that RPC.
Ownership is about clients and servers that own NetworkObjects, not NetworkObjects that own other NetworkObjects

Reply:
Nice catch, fixed!